src/AutofacSerilogIntegration/ContextualLoggingModule.cs

```diff
@@ -48,8 +50,17 @@ protected override void Load(ContainerBuilder builder)
 
                 return logger;
             })
-            .As<ILogger>()
-            .ExternallyOwned();
+            .As<ILogger>();
+
+            if (_dispose)
+            {
+                registration.SingleInstance()
```

Comment:
> I've only tested it locally on a web project, where it worked fine. It is true though that I may be too focused on my own use case.

👍 

The problem though, is that the container needs to return a different instance of `ILogger` depending on the type it's being injected into. If we use `SingleInstance()`, whichever class's `ILogger` is created first will be returned for all the other classes requesting it, too.